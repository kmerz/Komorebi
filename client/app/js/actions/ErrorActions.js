/*jshint esversion: 6 */
import AppDispatcher from '../dispatcher/AppDispatcher';
var ErrorActions = {
  addBoardErrors: (errors) => {
    AppDispatcher.dispatch({
      actionType: "ADD_BOARD_ERRORS",
      errors: errors
    });
  },
  removeBoardErrors: () => {
    AppDispatcher.dispatch({
      actionType: "REMOVE_BOARD_ERRORS"
    });
  },
  addColumnErrors: (errors) => {
    AppDispatcher.dispatch({
      actionType: "ADD_COLUMN_ERRORS",
      errors: errors
    });
  },
  removeColumnErrors: () => {
    AppDispatcher.dispatch({
      actionType: "REMOVE_COLUMN_ERRORS"
    });
  },
  addTaskErrors: (errors) => {
    AppDispatcher.dispatch({
      actionType: "ADD_TASK_ERRORS",
      errors: errors
    });
  },
  removeTaskErrors: () => {
    AppDispatcher.dispatch({
      actionType: "REMOVE_TASK_ERRORS"
    });
  },
  addStoryErrors: (errors) => {
    AppDispatcher.dispatch({
      actionType: "ADD_STORY_ERRORS",
      errors: errors
    });
  },
  removeStoryErrors: () => {
    AppDispatcher.dispatch({
      actionType: "REMOVE_STORY_ERRORS"
    });
  },
  addUserErrors: (errors) => {
    AppDispatcher.dispatch({
      actionType: "ADD_USER_ERRORS",
      errors: errors
    });
  },
  removeUserErrors: () => {
    AppDispatcher.dispatch({
      actionType: "REMOVE_USER_ERRORS"
    });
  },
  addDodErrors: (errors) => {
    AppDispatcher.dispatch({
      actionType: "ADD_DOD_ERRORS",
      errors: errors
    });
  },
  removeDodErrors: () => {
    AppDispatcher.dispatch({
      actionType: "REMOVE_DOD_ERRORS"
    });
  },
  addStoryIssueErrors: (errors) => {
    AppDispatcher.dispatch({
      actionType: "ADD_STORY_ISSUE_ERRORS",
      errors: errors
    });
  },
  removeStoryIssueErrors: () => {
    AppDispatcher.dispatch({
      actionType: "REMOVE_STORY_ISSUE_ERRORS"
    });
  }
};
module.exports = ErrorActions;
