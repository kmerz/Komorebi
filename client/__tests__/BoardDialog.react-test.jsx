import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { mount } from 'enzyme';


import FlatButton from 'material-ui/FlatButton';
import BoardDialog from '../app/js/board_dialog';
import Dialog from 'material-ui/Dialog';
import injectTapEventPlugin from 'react-tap-event-plugin';
import ReactTestUtils from 'react-addons-test-utils';
import ReactDOM from 'react-dom';
import ErrorActions from '../app/js/actions/ErrorActions'
injectTapEventPlugin();

it('Should render BoardDialog', () => {
  const wrapper = shallow(
      <BoardDialog open={false} />
    );
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('Should show Error when Error action is called', () => {
  const wrapper = mount(
    <BoardDialog open={true} />
  );
  expect(wrapper.state().error.board_name).toEqual("");

  ErrorActions.addBoardErrors({board_name: "Error!"});

  expect(wrapper.state().error.board_name).toEqual("Error!");
});
