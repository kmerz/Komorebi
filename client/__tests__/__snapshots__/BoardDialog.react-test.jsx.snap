exports[`test Should render BoardDialog 1`] = `
<Dialog
  actions={
    Array [
      <FlatButton
        disabled={false}
        keyboardFocused={true}
        label="Ok"
        labelPosition="after"
        labelStyle={Object {}}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        primary={true}
        secondary={false} />,
    ]
  }
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  modal={false}
  onRequestClose={[Function]}
  open={false}
  repositionOnUpdate={true}
  title="Add New Board">
  Add a name for the new Board
  <br />
  <TextField
    disabled={false}
    errorText=""
    floatingLabelFixed={false}
    fullWidth={false}
    hintText="Board Name"
    multiLine={false}
    onChange={[Function]}
    rows={1}
    type="text"
    underlineShow={true}
    value="" />
  <br />
  <br />
  Public or Private?
  <br />
  <Checkbox
    checked={false}
    checkedIcon={<ActionVisibilityOff />}
    disabled={false}
    label="Public"
    labelPosition="right"
    labelStyle={
      Object {
        "fontWeight": "normal",
      }
    }
    onCheck={[Function]}
    uncheckedIcon={<ActionVisibility />} />
</Dialog>
`;
